test/upgrade-e2e/post_upgrade_test.go

```diff
@@ -40,6 +40,20 @@ func TestClusterExtensionAfterOLMUpgrade(t *testing.T) {
 	t.Log("Wait for operator-controller deployment to be ready")
 	managerPod := waitForDeployment(t, ctx, "operator-controller-controller-manager")
 
+	t.Log("Start measuring leader election time")
```

Comment:
I think we need to be careful about how we measure the timing here. What we are measuring right now is the amount of time between:
-  the test detecting that the operator-controller deployment is finished, and
- how it takes for `watchPodLogsForSubstring(leaderElectionCtx, managerPod, "manager", leaderSubstrings...)` to return

This may correlate with the time taken for leader election, but it won't _necessarily_ correlate with it. E.g. let's say I upgrade the deployments, go out for lunch for 1h, come back and run the post upgrade test.

Maybe it would be better to extract the timestamp in the first and leader election log lines instead?